RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionLoading.swift

```diff
@@ -106,7 +114,60 @@ struct DappInteractionLoading: Sendable, FeatureReducer {
 						return .failure(error)
 					}
 					loggerGlobal.warning("Failed to fetch Dapps metadata, but since 'isDeveloperModeEnabled' is enabled we surpress the error and allow continuation. Error: \(error)")
-					return .success(.request(request))
+					return .success(.request(metadata))
+				}
+
+			}()
+
+			await send(.internal(.dappMetadataLoadingResult(result)))
+		}
+	}
+
+	private func deeplinkMetadataEffect(items: P2P.Dapp.Request.VerifyItems, metadata: P2P.Dapp.Request.Metadata) -> Effect<Action> {
+		.run { send in
+			guard
+				let wellKnownFile = try? await httpClient.fetchDappWellKnownFile(items.dappOrigin),
+				let dappDefinitionAddress = wellKnownFile.dApps.first?.dAppDefinitionAddress
+			else {
+				struct MissingDappDefinitionAddress: Error {}
+				throw MissingDappDefinitionAddress()
+			}
+
+			let result: TaskResult<DappMetadata> = await {
+				let isDeveloperModeEnabled = await appPreferencesClient.getPreferences().security.isDeveloperModeEnabled
+				do {
+					let cachedMetadata = try await cacheClient.withCaching(
+						cacheEntry: .dAppRequestMetadata(dappDefinitionAddress.address),
+						invalidateCached: { (cached: DappMetadata.DeepLink) in
+							guard
+								cached.name != nil,
+								cached.description != nil,
+								cached.thumbnail != nil
+							else {
+								/// Some of these fields were not set, fetch and see if they
+								/// have been updated since last time...
+								return .cachedIsInvalid
+							}
+							// All relevant fields are set, the cached metadata is valid.
+							return .cachedIsValid
+						},
```

Comment:
This logic is repeated with the on [ledgerMetadataEffect](https://github.com/radixdlt/babylon-wallet-ios/pull/1173/files#diff-54ac12251b3517f76c050913fc7dd30a8f6ef0bf9a1ad897d1d657835e86db3fR81). We may want to keep the useful info (`name`, `description` & `thumbnail`) in a new struct that both `Ledger` and `Deeplink` have, and implement a protocol so that this logic is done only once.. Wasn't entire sure if it was worthy so skipped for now.